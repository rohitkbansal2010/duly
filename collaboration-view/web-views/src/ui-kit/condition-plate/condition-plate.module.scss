@import "~@styles/colors.scss";

@mixin common-icon {
  display: flex;
  align-items: center;
  justify-content: center;
  box-sizing: border-box;
  width: 3.75rem;
  height: 3.75rem;
  padding: 0.75rem;
  border-radius: 100%;

  img {
    width: 100%;
    height: 100%;
  }
}

.conditionPlate {
  display: flex;
  align-items: center;
  padding: 1.25rem;
  background: white;
  border-radius: 0.5rem;

  &Icon {
    @include common-icon;
    flex-shrink: 0;
    background: lighten($dupage-light-blue, 40%);

    &Current {
      @include common-icon;
      background: lighten($dupage-magenta, 40%);
    }
  }

  &Data {
    margin-left: 1.25rem;
    word-break: break-word;

    &Name {
      margin: 0;
      font-weight: 800;
      font-size: var(--font-size-24px-4px);
      // line-height: 1.5rem;
    }

    &Date {
      font-size: var(--font-size-16px-4px);
      line-height: var(--line-height-24px-4px);
    }
  }
}
