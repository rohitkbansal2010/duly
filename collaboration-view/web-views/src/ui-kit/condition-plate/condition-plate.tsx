import { conditionsIcon, currentConditionsIcon } from '@icons';
import { formatMDYYYYDate } from '@utils';

import styles from './condition-plate.module.scss';

export type ConditionPlateProps = {
  title: string;
  date?: string;
  isCurrent?: boolean;
};

export const ConditionPlate = ({ title, date, isCurrent }: ConditionPlateProps) => 
  (
    <div className={styles.conditionPlate}>
      <div className={styles[`conditionPlateIcon${isCurrent ? 'Current' : ''}`]}>
        <img
          src={isCurrent ? currentConditionsIcon : conditionsIcon}
          alt="conditions icon"
        />
      </div>
      <div className={styles.conditionPlateData}>
        <h2 className={styles.conditionPlateDataName}>{title}</h2>
        <span className={styles.conditionPlateDataDate}>{date && formatMDYYYYDate(date)}</span>
      </div>
    </div>
  );
