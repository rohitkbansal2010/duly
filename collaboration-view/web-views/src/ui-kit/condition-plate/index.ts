export { ConditionPlate } from './condition-plate';
