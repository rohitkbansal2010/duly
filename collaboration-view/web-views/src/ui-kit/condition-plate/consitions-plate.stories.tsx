import { ConditionPlate } from './condition-plate';

export default {
  title: 'UI-KIT / ConditionPlate',
  component: ConditionPlate,
};

export const Example = () => 
  (
    <div style={{ background: '#ddd', width: '600px', padding: 40 }}>
      <ConditionPlate isCurrent title="Test condition" date="12/12/12" />
    </div>
  );
