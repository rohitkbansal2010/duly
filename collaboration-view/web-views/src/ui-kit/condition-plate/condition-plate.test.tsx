import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { conditionsIcon, currentConditionsIcon } from '@icons';
import { healthConditionsMockData } from '@mock-data';
import { ConditionPlate } from '@ui-kit/condition-plate';
import { formatMDYYYYDate } from '@utils';

const { currentHealthConditions } = healthConditionsMockData;
const mockProps = currentHealthConditions[0];

describe('HealthConditionsPlate', () => {
  let component: ReturnType<typeof shallow>;

  beforeEach(() => {
    component = shallow(<ConditionPlate {...mockProps} />);
  });

  it('component should match to snapshot', () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  it('component should render right data', () => {
    const { title, date } = mockProps;
  
    expect(component.text().includes(title)).toBeTruthy();
    expect(component.text().includes(formatMDYYYYDate(date!))).toBeTruthy();
  });

  it('component should render right icon', () => {
    component = shallow(<ConditionPlate {...mockProps} isCurrent={true} />);

    expect(component.findWhere(n =>
      n.type() === 'img' && n.props().src === currentConditionsIcon)).toHaveLength(1);

    component = shallow(<ConditionPlate {...mockProps} isCurrent={false} />);

    expect(component.findWhere(n =>
      n.type() === 'img' && n.props().src === conditionsIcon)).toHaveLength(1);
  });
});
