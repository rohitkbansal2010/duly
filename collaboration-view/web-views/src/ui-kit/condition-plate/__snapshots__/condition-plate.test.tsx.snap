// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HealthConditionsPlate component should match to snapshot 1`] = `
<div
  className="conditionPlate"
>
  <div
    className="conditionPlateIcon"
  >
    <img
      alt="conditions icon"
      src="conditions-icon.svg"
    />
  </div>
  <div
    className="conditionPlateData"
  >
    <h2
      className="conditionPlateDataName"
    >
      Pneumonia
    </h2>
    <span
      className="conditionPlateDataDate"
    >
      2/16/2022
    </span>
  </div>
</div>
`;
